### PR TITLE
Remove Book To Nomis code

### PR DESCRIPTION
**Pull request by KPobeeNorris**
Following a decision to no longer continue with the development and
roll out of Book To Nomis we are going to remove the code to reduce
cofusion and complexity.

The first step was to remove the configuration and feature flags related
to Book To Nomis and get all tests passing again.  We then removed all code related to the Book To Nomis feature, removed references to it from the readme/docs, and removed the now unused VCR cassettes .

### CODE REVIEW

File: app/models/visit.rb [hunk 1]
```diff
@@ -10,7 +10,6 @@ class Visit < ApplicationRecord
   has_many :messages, dependent: :destroy
   has_one :rejection, dependent: :destroy, inverse_of: :visit
   has_one :cancellation, dependent: :destroy
-  has_one :visit_order, dependent: :destroy
```
Comment (TobyRet) [inline app/models/visit.rb]: Let's add this back in and remove in another PR when we clean up the db

File: app/models/visit_order.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
+# :nocov:
+# TODO: Remove
```
Comment (TobyRet) [inline app/models/visit_order.rb]: To be done in another PR

File: app/models/visit_order/priviledged.rb [hunk 3]
```diff
@@ -1,2 +0,0 @@
-class VisitOrder::Priviledged < VisitOrder
```
Comment (TobyRet) [inline app/models/visit_order/priviledged.rb]: Let's add this back in and remove in another PR when we clean up the db

File: app/models/visit_order/unsupported.rb [hunk 4]
```diff
@@ -1,2 +0,0 @@
-class VisitOrder::Unsupported < VisitOrder
```
Comment (TobyRet) [inline app/models/visit_order/unsupported.rb]: Let's add this back in and remove in another PR when we clean up the db

File: app/services/nomis/client.rb [hunk 5]
```diff
@@ -100,28 +92,28 @@ def request(method, route, params, idempotent:, options: {})
...
-    def http_method_expects(method)
-      if method == :get
-        [200]
-      else
-        # TODO: Change 400 to 422 when api is updated.
-        # Currently it returns 400 for validation errors.
-        [200, 400, 409]
-      end
-    end
+    # def http_method_expects(method)
```
Comment (TobyRet) [inline app/services/nomis/client.rb]: My bad - should be removed

File: app/services/nomis/client.rb [hunk 6]
```diff
@@ -100,28 +92,28 @@ def request(method, route, params, idempotent:, options: {})
...
     def params_options(method, params)
       return {} if params.empty?
 
-      if [:post, :patch].include?(method)
-        {
-          body: params.to_json,
-          headers: { 'Content-Type' => JSON_MIME_TYPE }
-        }
-      else
+      # if [:post, :patch].include?(method)
```
Comment (TobyRet) [inline app/services/nomis/client.rb]: Same here

File: app/views/prison/visits/_book_to_nomis_opt_in.html.erb [hunk 7]
```diff
@@ -1,48 +0,0 @@
...
-      <hr>
-      <div class="grid-row" id="opt-out-check">
-
-        <div class="form-group column-two-thirds">
-          <fieldset>
-            <legend class="push-bottom--half">
-              <h2 class="bold-medium">
-                Copy this visit to NOMIS?
-              </h2>
-              <p><%= t(".book_to_nomis_opted_in_by_default") %></p>
```
Comment (TobyRet) [inline app/views/prison/visits/_book_to_nomis_opt_in.html.erb]: prison_views.yml needs to have related copy deleted also

File: config/initializers/types.rb [hunk 8]
```diff
@@ -23,4 +23,3 @@
 ActiveModel::Type.register(:restriction_list,                RestrictionListType)
 ActiveModel::Type.register(:visitor,                         VisitorType)
 ActiveModel::Type.register(:visitor_list,                    VisitorListType)
-ActiveModel::Type.register(:visit_order,                     VisitOrderType)
```
Comment (TobyRet) [inline config/initializers/types.rb]: Keep in this PR and remove in the next PR

File: lib/types/visit_order_type.rb [hunk 9]
```diff
@@ -1,7 +0,0 @@
-class VisitOrderType < ActiveModel::Type::Value
```
Comment (TobyRet) [inline lib/types/visit_order_type.rb]: Keep in this PR and remove in the next one

File: spec/controllers/prison/cancellations_controller_spec.rb [hunk 10]
```diff
@@ -65,4 +65,36 @@
...
+      }
+    end
+
+    before do
+      login_user(create(:user), current_estates: [estate])
+
+      allow(GATracker).to receive(:new).and_return(ga_tracker)
+    end
+
+    context 'when the visit is already cancelled' do
```
Comment (TobyRet) [inline spec/controllers/prison/cancellations_controller_spec.rb]: Context incorrectly named

File: spec/controllers/prison/cancellations_controller_spec.rb [hunk 11]
```diff
@@ -65,4 +65,36 @@
...
+      it 'redirect to the visit show page setting the already cancelled flash message' do
```
Comment (TobyRet) [inline spec/controllers/prison/cancellations_controller_spec.rb]: Test-case incorrectly named

File: spec/models/visit_order_spec.rb [hunk 12]
```diff
@@ -1,5 +0,0 @@
-require 'rails_helper'
-
-RSpec.describe VisitOrder, type: :model do
```
Comment (TobyRet) [inline spec/models/visit_order_spec.rb]: Reinstate and remove in next PR

File: app/assets/javascripts/modules/moj.closedRestriction.js [hunk 13]
```diff
@@ -16,7 +16,6 @@
     cacheEls: function(){
       this.$el = $(this.el);
       this.$closedEl = $('#closed-visits');
-      this.$nomisEl = $('#book_to_nomis_opt_in');
```
[the remark this answers is not in the document]
Comment (KPobeeNorris) [reply app/assets/javascripts/modules/moj.closedRestriction.js]: @chrishanes - thanks for this; all removed!

File: config/routes.rb [hunk 14]
```diff
@@ -33,10 +33,6 @@
     namespace :prison do
       resources :visits, only: %i[show update] do
         resource :cancellations, only: :create
-        member do
```
Comment (chrishanes) [inline config/routes.rb]: I think we need this as well for normal cancellations?
